### PR TITLE
#451 Make dependencies in squbs-ext and squbs-pattern mostly optional

### PR DESCRIPTION
**Pull request by akara**
... dependent on the functionality to use.

Thanks for your pull request.  Please review the following guidelines.

- [X] Title includes issue id.
- [X] Description of the change added.
- [X] Commits are squashed.
- [ ] Tests added.
- [X] Documentation added/updated.
- [X] Also please review [CONTRIBUTING.md](https://github.com/paypal/squbs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: docs/marshalling.md [hunk 1]
```diff
@@ -17,11 +17,32 @@ This document discusses the marshallers and unmarshallers provided by squbs, and
...
+"com.fasterxml.jackson.core" % "jackson-databind" % jacksonVersion
 
 ```
-"org.squbs" %% "squbs-ext" % squbsVersion
-```
+
+The followings are optional components depending on your marshalling formats and libraries you want to use. They may be combined to support multiple formats.
+
+```scala
+  // To use json4s native...
```
Comment (anilgursel) [inline docs/marshalling.md]: Just formatting, but it would read better if there is no space on the left.

File: squbs-ext/build.sbt [hunk 2]
```diff
@@ -13,16 +13,16 @@ testOptions in Test ++= Seq(
...
-  "com.typesafe.akka" %% "akka-http-core" % akkaHttpV % "provided",
-  "de.heikoseeberger" %% "akka-http-json4s" % "1.11.0",
-  "de.heikoseeberger" %% "akka-http-jackson" % "1.11.0",
-  "org.json4s" %% "json4s-native" % "3.5.0",
-  "org.json4s" %% "json4s-jackson" % "3.5.0",
-  "com.fasterxml.jackson.core" % "jackson-core" % jacksonV,
-  "com.fasterxml.jackson.core" % "jackson-databind" % jacksonV,
-  "com.fasterxml.jackson.module" %% "jackson-module-scala" % "2.8.4",
-  "com.fasterxml.jackson.module" % "jackson-module-parameter-names" % jacksonV,
+  "com.typesafe.akka" %% "akka-stream" % akkaV % "optional",
```
Comment (anilgursel) [inline squbs-ext/build.sbt]: 👍

File: squbs-ext/build.sbt [hunk 3]
```diff
@@ -13,16 +13,16 @@ testOptions in Test ++= Seq(
 libraryDependencies ++= Seq(
   "org.scala-lang.modules" %% "scala-java8-compat" % "0.7.0",
   "io.dropwizard.metrics" % "metrics-core" % metricsV,
-  "com.typesafe.akka" %% "akka-stream" % akkaV,
-  "com.typesafe.akka" %% "akka-http-core" % akkaHttpV % "provided",
```
Comment (sebady) [inline squbs-ext/build.sbt]: Doesn't `provided` already imply optionality here?

Comment (anilgursel) [inline squbs-ext/build.sbt]: `provided` means, someone needs to provide it though.  `optional` means, it does not have to be provided.  It need to be provided only if that feature will be used.

Comment (akara) [inline squbs-ext/build.sbt]: Thanks for explaining. Yes, I looked up the docs and `optional` seems to match the semantics the best. The `provided` is really designed for uses such as containers providing the Servlet API + implementation.

**Pull request merged**